ansible/roles-infra/infra-ibm-rg-create/tasks/main.yml

```diff
@@ -1,45 +1,46 @@
 ---
-# TODO Use ansible modules
-# TODO Use json_query
 - name: Getting the resource-group list
-  command: >-
+  command: >
     ibmcloud resource groups --output json
   register: r_rg_list
 
 - debug:
     var: r_rg_list.stdout | from_json
 
-- name: Check if resource group already exists
-  set_fact:
-    resource_group_exists: false
-
-- name: Check if resource group already exists
-  when: (r_rg_list.stdout | from_json) | json_query(rg_query) | length > 0
-  set_fact:
-    resource_group_exists: true
+- name: Create RG if does not exist
+  when: (r_rg_list.stdout | from_json) | json_query(rg_query) | length == 0
+  command: >-
+    ibmcloud resource group-create {{ guid }}-rg
   vars:
     rg_query: >
       [?name=='{{ guid }}-rg']
+  tags:
+    - resource_groups
+
+- name: Check if resource group already exists
+  set_fact:
+    resource_group_exists: false
 
 - debug:
     var: resource_group_exists
 
-- name: Get the ID of resource group if it exists
-  when: resource_group_exists
-  set_fact:
-    resource_group_id: "{{ (r_rg_list.stdout | from_json) | json_query(rg_id_query) }}"
-  vars:
-    rg_id_query: >
-      [?name=='{{ guid }}-rg'].id|[0]
+# I don't see why do we need this
```

Comment:
ID is more precise than name and I'm not sure if we will use it again later. IBM is not consistent with their tools and whether they want a name or ID. It doesn't hurt to have, so I am going to leave it.

Reply:
Agreed, commit will follow.